### PR TITLE
Upgrade rubocop to 0.36

### PR DESCRIPTION
**Pull request by jawshooah**
Also fix offenses introduced by the upgrade.

[Review on Reviewable](https://reviewable.io/reviews/nevir/rubocop-rspec/60)

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "vendor/rubocop"]
```
Comment (andyw8) [inline .gitmodules]: Did you intend to commit this?

File: spec/project_spec.rb [hunk 2]
```diff
@@ -5,12 +5,12 @@
     let(:cop_names) do
       Dir.glob(File.join(File.dirname(__FILE__), '..', 'lib', 'rubocop', 'cop',
                          'rspec', '*.rb'))
-        .map do |file|
-          cop_name = File.basename(file, '.rb')
-            .gsub(/(^|_)(.)/) { Regexp.last_match(2).upcase }
+         .map do |file|
+        cop_name = File.basename(file, '.rb')
```
Comment (andyw8) [inline spec/project_spec.rb]: Should this be indented?

File: .gitmodules (hunk 1 above)
Comment (jawshooah) [inline .gitmodules]: Yes, and I meant to make a change to the README as well. Makes much more sense to just add rubocop as a submodule than to require the user to explicitly `git clone` the latest master and hope nothing breaks.

File: spec/project_spec.rb (hunk 2 above)
Comment (jawshooah) [inline spec/project_spec.rb]: Yup, forgot to update `.rubocop.yml` with the new `MutilineMethodCallIndentation` cop.

File: lib/rubocop/cop/rspec/describe_class.rb [hunk 3]
```diff
@@ -24,7 +25,7 @@ class DescribeClass < Cop
         FEATURE_PAIR = s(:pair, s(:sym, :type), s(:sym, :feature))
 
         MESSAGE = 'The first argument to describe should be the class or ' \
-                  'module being tested.'
+                  'module being tested.'.freeze
```
Comment (andyw8) [inline lib/rubocop/cop/rspec/describe_class.rb]: I may have misunderstood it, but doesn't setting `frozen_string_literal` mean that there's no need to call `.freeze` on each string?

File: .gitmodules (hunk 1 above)
Comment (andyw8) [inline .gitmodules]: I see. You might want to make that a separate PR since it's not directly related to the RuboCop upgrade.

File: lib/rubocop/cop/rspec/describe_class.rb (hunk 3 above)
Comment (jawshooah) [inline lib/rubocop/cop/rspec/describe_class.rb]: Yes, but [rubocop currently registers an offense regardless](https://github.com/bbatsov/rubocop/pull/2634).

File: .gitmodules (hunk 1 above)
Comment (jawshooah) [inline .gitmodules]: Fair point; I'll split it out.

Comment (jawshooah) [inline .gitmodules]: Done in #61.

File: lib/rubocop/cop/rspec/describe_class.rb (hunk 3 above)
Comment (andyw8) [inline lib/rubocop/cop/rspec/describe_class.rb]: Perhaps it would be better to disable that check until the issue is fixed in RuboCop? Otherwise a bunch of lines are going to be churned (by adding `.freeze` then removing in later).

Comment (jawshooah) [inline lib/rubocop/cop/rspec/describe_class.rb]: Done.

**Comment by davidakachaos** [comment]
:+1: @jawshooah could you look into the 'codeclimate-test-reporter' error in the test?

**Comment by jawshooah** [comment]
That error has been around ever since RuboCop switched from Coveralls to CodeClimate, nothing to do with the changes in this PR.

**Comment by davidakachaos** [comment]
Great! In that case ping @nevir :smile:

**Comment by shaneog** [comment]
My PR (#63) should fix the Travis issues with CodeClimate, but it requires both PR's by @jawshooah to be merged first.

**Comment by nijikon** [comment]
What is the status of this, when can we merge?

**Comment by lucascaton** [comment]
Any news on this one?

**Comment by jawshooah** [comment]
Waiting on @nevir, looks like.

**Comment by bogdanRada** [comment]
+1

**Comment by aried3r** [comment]
Now there's Rubocop 0.37 https://github.com/bbatsov/rubocop/blob/master/CHANGELOG.md#0370-04022016

**Comment by nijikon** [comment]
Yup, I will merge this today, and go through other tickets. Then update to 0.37.0.

**Comment by DexterTheDragon** [comment]
@nijikon Will you be pushing a new version to rubygems?

**Comment by nijikon** [comment]
@DexterTheDragon yes, working on it right now
